velox/functions/lib/DateTimeFormatter.h

```diff
@@ -0,0 +1,371 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <string>
+#include <vector>
+#include "velox/common/base/Exceptions.h"
+#include "velox/type/Timestamp.h"
+
+namespace facebook::velox::functions {
+
+static constexpr size_t kStructReserveSize{8};
+static constexpr size_t kLiteralReserveSize{64};
+
+enum class DateTimeFormatSpecifier : uint8_t {
```

Comment:
These are an item-by-item copy of what Joda provides today. Does this also have a 1:1 mapping to what we need for Mysql?